File: Anole/HTMLFontElement.vb
Imports System
Imports System.Xml
Imports System.Drawing
Imports System.Windows.Forms

Namespace Anole
    ''' <summary>
    ''' Summary description for HTMLFontElement.
    ''' </summary>
    Public Class HTMLFontElement
        Inherits HTMLElement
        Public Sub New(ByRef document As HTMLDocument, parent As HTMLElement, xnode As XmlNode)
            Initialize(document, parent, xnode)
            
            ParseForeColor(xnode)
            
            Dim fname As String = Me.ParseString(xnode, "face", Me.Font.Name)
            Dim fsize As Single = Me.ParseFloat(xnode, "size", Me.Font.Size)
            
            If (fname.CompareTo(Me.Font.Name) <> 0) OrElse (fsize <> Font.Size) Then
                'Create a new font
                Me.Font = New Font(fname, fsize, Me.Font.Style)
            End If
            
            ParseChildren(xnode)
        End Sub
    End Class
    
    Public Class HTMLBoldElement
        Inherits HTMLElement
        Public Sub New(ByRef document As HTMLDocument, parent As HTMLElement, xnode As XmlNode)
            Initialize(document, parent, xnode)
            Dim fname As String = Me.Font.Name
            Dim fsize As Single = Me.Font.Size
            Dim fs As FontStyle = Me.Font.Style
            fs = fs Or FontStyle.Bold
            Me.Font = New Font(fname, fsize, fs)
            ParseChildren(xnode)
        End Sub
    End Class
    
    Public Class HTMLUnderscoreElement
        Inherits HTMLElement
        Public Sub New(ByRef document As HTMLDocument, parent As HTMLElement, xnode As XmlNode)
            Initialize(document, parent, xnode)
            Dim fname As String = Me.Font.Name
            Dim fsize As Single = Me.Font.Size
            Dim fs As FontStyle = Me.Font.Style
            fs = fs Or FontStyle.Underline
            Me.Font = New Font(fname, fsize, fs)
            ParseChildren(xnode)
        End Sub
    End Class
    
    Public Class HTMLItalicElement
        Inherits HTMLElement
        Public Sub New(ByRef document As HTMLDocument, parent As HTMLElement, xnode As XmlNode)
            Initialize(document, parent, xnode)
            Dim fname As String = Me.Font.Name
            Dim fsize As Single = Me.Font.Size
            Dim fs As FontStyle = Me.Font.Style
            fs = fs Or FontStyle.Italic
            Me.Font = New Font(fname, fsize, fs)
            ParseChildren(xnode)
        End Sub
    End Class
End Namespace

File: Anole/HTMLFontElement.cs
using System;
using System.Xml;
using System.Drawing;
using System.Windows.Forms;

namespace Anole
{
	/// <summary>
	/// Summary description for HTMLFontElement.
	/// </summary>
	public class HTMLFontElement : HTMLElement
	{
		public HTMLFontElement(ref HTMLDocument document,HTMLElement parent,XmlNode xnode)
		{
			Initialize(ref document, parent, xnode);

			ParseForeColor(xnode);

			string fname=this.ParseString(xnode,"face",this.Font.Name);
			float fsize=this.ParseFloat(xnode,"size",this.Font.Size);
			
			if(
				(fname.CompareTo(this.Font.Name)!=0)||
				(fsize!=Font.Size)
				)
			{
				//Create a new font
				this.Font=new Font(fname,fsize,this.Font.Style);
			}

			ParseChildren(xnode);
		}
	}

	public class HTMLBoldElement : HTMLElement
	{
		public HTMLBoldElement(ref HTMLDocument document, HTMLElement parent, XmlNode xnode)
		{
			Initialize(ref document, parent, xnode);
			string fname=this.Font.Name;
			float fsize=this.Font.Size;
			FontStyle fs=this.Font.Style;
			fs|=FontStyle.Bold;
			this.Font=new Font(fname,fsize,fs);
			ParseChildren(xnode);
		}
	}

	public class HTMLUnderscoreElement : HTMLElement
	{
		public HTMLUnderscoreElement(ref HTMLDocument document, HTMLElement parent, XmlNode xnode)
		{
			Initialize(ref document, parent, xnode);
			string fname=this.Font.Name;
			float fsize=this.Font.Size;
			FontStyle fs=this.Font.Style;
			fs|=FontStyle.Underline;
			this.Font=new Font(fname,fsize,fs);
			ParseChildren(xnode);
		}
	}

	public class HTMLItalicElement : HTMLElement
	{
		public HTMLItalicElement(ref HTMLDocument document, HTMLElement parent, XmlNode xnode)
		{
			Initialize(ref document, parent, xnode);
			string fname=this.Font.Name;
			float fsize=this.Font.Size;
			FontStyle fs=this.Font.Style;
			fs|=FontStyle.Italic;
			this.Font=new Font(fname,fsize,fs);
			ParseChildren(xnode);
		}
	}
}
